### PR TITLE
Move MigrationExtractors to Activity

### PR DESCRIPTION
**Pull request by MechaRyRy**
## Problem
We are unable to do a release because our `Migration` implementation is very specific and does not allow clients to specify where they retrieve their V1 data from. 

## Solution 
Move the `MigrationExtractor`s to the `MainActivity` and have the `MigrationJob` read through the `List<Migration>` and add them to the `v2` database. When the `COMPLETE` status is emitted, delete the v1 database.

## Follow-up PRs **discussion needed**
This completes the whole flow but in a very simplistic manner. We no longer emit the deleting v1 database and it might be more appropriate to return a list of completed migrations so the clients can act on them if necessary.  We need to remember to remove the files for the partial migrations, this can be done either internally or left to the client.

### CODE REVIEW

File: demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java [hunk 1]
```diff
@@ -119,12 +127,37 @@ protected void onCreate(Bundle savedInstanceState) {
 
     private final MigrationCallback migrationCallback = migrationStatus -> {
         if (migrationStatus.status() == MigrationStatus.Status.COMPLETE) {
+            File databasePath = getDatabasePath("downloads.db");
```
Comment (danybony) [inline demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java]: Extract this to a constant, given it's used twice

Comment (danybony) [reply demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java]: "downloads.db" I mean

**Review by danybony — COMMENTED** [review]
Ok, now it's definitely more flexible for the client, maybe even too much? Don'y you think it's possible to enable some sort of default mechanism?

**Comment by MechaRyRy** [comment]
@danybony what do you mean by a default mechanism? The `v1` db could be used in so many different ways it would be very difficult to predict what fields the clients are using to represent our `v2` concepts 😕 of course, if you have ideas then I am happy to try them 😄

**Comment by danybony** [comment]
No @Mecharyry you are right, I was just discussing the possibility of having something that could be used to cover the most common scenario.
But as you stated, it's not possible to predict how the library was used in v1 by clients, so good with giving them flexibility

**Review by danybony — APPROVED** [review]

**Comment by MechaRyRy** [comment]
Going for a different approach.

**Pull request closed without merging**